staging/src/k8s.io/client-go/rest/request_test.go

```diff
@@ -488,6 +489,14 @@ func TestURLTemplate(t *testing.T) {
 			ExpectedFullURL:  "http://localhost/some/base/url/path/pre1/namespaces/namespaces/namespaces/namespaces/namespaces/namespaces/finalize",
 			ExpectedFinalURL: "http://localhost/%7Bprefix%7D",
 		},
+		{
+			// dynamic client with core group + namespace + resourceResource (with name) where baseURL is a single /
```

Comment:
adding client go dynamic client expert @roycaihw , would you please help check the correctness of the dynamic client format?

Reply:
the problem also occurs when `baseURL` is something like `url.Parse("base/url")`. It would be great if you can also handle that case. LGTM otherwise

I don't see if we put restriction on what `baseURL` can be, but that's probably out of scope for this function